File: spring-suite-ui/src/app/models/event.ts
export interface EventInvitee {
  email: string;
  status: 'PENDING' | 'ACCEPTED' | 'DECLINED';
}

export interface Event {
  id: number;
  title: string;
  description: string;
  location: string | null;
  startTime: string;
  endTime: string;
  owner: string;
  attendees: string[];
  invitees: EventInvitee[];
  createdAt: string;
}

export interface EventRequest {
  title: string;
  description?: string;
  location?: string;
  startTime: string;
  endTime: string;
  attendees?: string[];
  invitees?: EventInvitee[];
}

export interface InviteeRequest {
  email: string;
  status?: 'PENDING' | 'ACCEPTED' | 'DECLINED';
}

File: calendar-service/src/main/java/fr/matteofierquin/calendarservice/model/Event.java
package fr.matteofierquin.calendarservice.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "events")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(length = 1000)
    private String description;

    private String location;

    @Column(nullable = false)
    private LocalDateTime startTime;

    @Column(nullable = false)
    private LocalDateTime endTime;

    @Column(nullable = false)
    private String owner;  // Username of event creator

    @ElementCollection
    @CollectionTable(name = "event_attendees", joinColumns = @JoinColumn(name = "event_id"))
    @Column(name = "attendee_username")
    @Builder.Default
    private List<String> attendees = new ArrayList<>();

    @ElementCollection
    @CollectionTable(name = "event_invitees", joinColumns = @JoinColumn(name = "event_id"))
    @Builder.Default
    private List<EventInvitee> invitees = new ArrayList<>();

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        validate();
    }

    @PreUpdate
    protected void onUpdate() {
        validate();
    }

    protected void validate() {
        if (endTime != null && startTime != null && !endTime.isAfter(startTime)) {
            throw new IllegalArgumentException("End time must be after start time");
        }
    }
}